Sage/carbon
# docs(contributing): update existing docs to be inline with present ways of working

### Proposed behaviour


- Correct name for `DefinitionList`'s test story - so it does not appear at the bottom of the story list in storybook when run locally:
![Screenshot 2022-10-19 at 12 02 09](https://user-images.githubusercontent.com/18368713/196673710-c29580ae-8e33-4873-bb7c-f2ab9f667f81.png)
- Rewrite existing docs on contributing to be inline with our current workflow - with the target audience being **developers who need concise details on how to get started**.
- Move details on legacy cypress E2E test to a separate file.
- Add new PR checklist item for checking documentation.
- In storybook docs, add `How to contribute` page and move `Development roadmap` page under new `Contributing` section. `How to contribute` provides a link to `CONTRIBUTING.md` on github.

### Current behaviour


- Test story for `DefinitionList` has inconsistent name - so it appears in a separate dropdown category in the story list when storybook is run locally:
![Screenshot 2022-10-19 at 12 03 12](https://user-images.githubusercontent.com/18368713/196673903-7c9e7e79-48da-43b2-853a-63995a5611d0.png)

### Checklist




#### QA


### Additional context



### Testing instructions

- Double check for errors in grammar and ambiguous statements
- Suggest possible improvements to make docs clearer for new contributors
